src/core/trtserver.h

```diff
@@ -465,32 +466,127 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceManagerNew(
+    TRTSERVER_InferenceRequestHeader* request_header, const char* input_name,
+    const int64_t* dims, uint64_t dim_count, uint64_t batch_byte_size);
+
+/// Request the raw data of the specified output to be returned for requests
+/// associated with the request header.
+/// \param request_header The request header object.
+/// \param output_name The name of the output.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestHeaderAddOutputRaw(
+    TRTSERVER_InferenceRequestHeader* request_header, const char* output_name);
+
+/// Request the classification values of the specified output to be returned
+/// for requests associated with the request header.
+/// \param request_header The request header object.
+/// \param output_name The name of the output.
+/// \param count Indicates how many classification values should be returned
+/// for the output. The 'count' highest priority values are returned.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestHeaderAddOutputCls(
+    TRTSERVER_InferenceRequestHeader* request_header, const char* output_name,
+    uint32_t count);
+
+/// Delete an inference request header object.
+/// \param request_header The request header object.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestHeaderDelete(
+    TRTSERVER_InferenceRequestHeader* request_header);
+
 /// TRTSERVER_InferenceRequestProvider
 ///
 /// Object representing the request provider for an inference
 /// request. The request provider provides the meta-data and input
 /// tensor values needed for an inference.
 ///
 
-/// Create a new inference request provider object. The request header
-/// protobuf must be serialized and provided as a base address and a
-/// size, in bytes.
+/// Create a new inference request provider object. The caller retains ownership
+/// of 'request_header' but may release it by calling
+/// TRTSERVER_InferenceRequestHeaderDelete once this function returns.
+/// However, the values in 'request_header' must not be modified until the
+/// request provider object can be deleted.
 /// \param request_provider Returns the new request provider object.
 /// \param server the inference server object.
-/// \param model_name The name of the model that the inference request
-/// is for.
-/// \param model_version The version of the model that the inference
-/// request is for, or -1 to select the latest (highest numbered)
-/// version.
-/// \param request_header_base Pointer to the serialized request
-/// header protobuf.
-/// \param request_header_byte_size The size of the serialized request
-/// header in bytes.
+/// \param request_header The request header object for this inference request.
 /// \return a TRTSERVER_Error indicating success or failure.
 TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestProviderNew(
```

Comment:
Can't remove this API yet as that would break backwards compatibility too severely. So just keep this one as it is but mark it as deprecated and point to a new TRTSERVER_InferenceRequestProviderNewV2(TRTSERVER_InferenceRequestProvider** request_provider, TRTSERVER_Server* server);